File: Python/Algorithm/BIN_to_DEC.py
# Input is a Binary String:
# Complexity: O(n)

# Method 1: Conversion from Binary String
def str(s):
    n = 0
    for i in s:
        n = (n << 1) | (int(i) - 0)
    return n

# Method 2: Direct Method using Python's built-in functionality
s = "1011"
n = int(s, 2)  # Convert binary string to integer
print(n)  # Output: 11

# Equivalent to binary literals
a = 0b1011
print(a)  # Output: 11

File: CPP/Algorithm/BIN_to_DEC.cpp
// Input is a Binary String: O(n)

int64_t bin(auto &s) {
    int64_t n = 0;
    for (auto &c : s) {
        n <<= 1; n |= (c - '0');
    }
    return n;
}

// Input is a Decimal Number: O(log n)

int64_t bin(auto n) {
    int64_t b = 0;
    while (n) {
        b <<= 1; b |= (n & 1); n >>= 1;
    }
    return b;
}

// Built-in Method: Complexity: O(1)
    string s = "1011";
    int64_t n = stoi(s, 0, 2);
    cout << n; // 11

// OR,
    int64_t a = 0b1011;
    cout << a; // 11